baseapp/msg_service_router.go

```diff
@@ -43,7 +43,7 @@ func (msr *MsgServiceRouter) SetCircuit(cb CircuitBreaker) {
 }
 
 // MsgServiceHandler defines a function type which handles Msg service message.
-type MsgServiceHandler = func(ctx sdk.Context, req sdk.Msg) (*sdk.Result, error)
+type MsgServiceHandler = func(ctx *sdk.Context, req sdk.Msg) (*sdk.Result, error)
```

Comment:
This also also an anti-pattern unfortunately. A context isn't meant to be mutable. Let's setup a time to chat syncchronously and see if we can derive a clean solution?

Reply:
another option would be to provide two interfaces for this. The existing one (maybe renamed to `MsgServiceHandlerWithContextReset` or similar?) and a new `MsgServiceHandlerPreserveContext`

Reply:
actually... we don't even need to modify the interface, just add a variable to control whether this is reset in the caller or inside the function